File: solutions/3651-transformed-array/transformed-array.c
int* constructTransformedArray(int* nums, int numsSize, int* returnSize) {
    *returnSize = numsSize;
    int* res = (int*)malloc(numsSize * sizeof(int));
    for (int i = 0; i < numsSize; i++) {
        res[i] = nums[((i + nums[i]) % numsSize + numsSize) % numsSize];
    }
    return res;
}

File: solutions/3651-transformed-array/transformed-array.ts
function constructTransformedArray(nums: number[]): number[] {
    const n = nums.length;
    const res: number[] = new Array(n);
    for (let i = 0; i < n; i++) {
        res[i] = nums[(((i + nums[i]) % n) + n) % n];
    }
    return res;
}